plugins/hashsum_download/server/__init__.py

```diff
@@ -118,7 +123,37 @@ def _getFirstFileByHash(self, algo, hash, user=None):
         return None
 
 
+def _computeHash(event):
+    """
+    Computes all supported checksums on a given file. Downloads the
+    file data and stream-computes all required hashes on it, saving
+    the results in the file document.
+    """
+    file = event.info['file']
+
+    toCompute = SUPPORTED_ALGORITHMS - set(file)
```

Comment:
Thanks for the clarification.

I see the `file` you use is from https://github.com/girder/girder/blob/41b3e3be68df34e55fbc88676bb81540d1cbfc43/girder/models/upload.py#L228-L241

And it contains keys like these ones ... 

```
{
    "_id" : ObjectId("5906bfba4154cd7dfbe47cfe"),
    "relpath" : "/trigirderdemo-assetstore/77/47/7747937d98b34aaab0a54e96395ddfc1",
    "mimeType" : "application/octet-stream",
    "assetstoreId" : ObjectId("590613464154cd7dfbe47cf7"),
    "size" : 8794800,
    "creatorId" : ObjectId("574305a44154cd4e67c1d835"),
    "dicom" : {},
    "exts" : [ 
        "obj"
    ],
    "s3Key" : "77/47/7747937d98b34aaab0a54e96395ddfc1",
    "itemId" : ObjectId("5906bf884154cd7dfbe47cfc"),
    "created" : ISODate("2017-05-01T04:55:22.129Z"),
    "name" : "baby_dragon.obj"
}
```